Feat: Global Color Schemes

kaeizen opened the pull request.

kaeizen committed:
init

Arukuen committed:
feat: inspector-sub-header for new global settings

Arukuen committed:
feat: inspector-sub-header for new global settings

kaeizen committed:
working responsive in editor and frontend

kaeizen committed:
Merge branch 'develop' into feat/global-block-layout

# Conflicts:
#	src/dynamic-breakpoints.php

kaeizen committed:
hover state

kaeizen committed:
hover states

Arukuen committed:
fix: use onClick properly

Arukuen committed:
fix: use onClick properly

Arukuen committed:
feat: global font pairs

Arukuen committed:
feat: global font pairs

Arukuen committed:
fix: use new scraped font pairs; adjust code to simpler json structure

Arukuen committed:
fix: use new scraped font pairs; adjust code to simpler json structure

Arukuen committed:
fix: make scrolling to current font pair independent from current view

Arukuen committed:
fix: make scrolling to current font pair independent from current view

Arukuen committed:
fix: rename stackable_font_pair_name to stackable_selected_font_pair

Arukuen committed:
fix: rename stackable_font_pair_name to stackable_selected_font_pair

Arukuen committed:
fix: rename stackable_font_pair_name to stackable_selected_font_pair, reset previous commit

Arukuen committed:
fix: rename stackable_font_pair_name to stackable_selected_font_pair, reset previous commit

Arukuen committed:
fix: move bulk of code to FontPairPicker, restructure with consistent naming

Arukuen committed:
fix: move bulk of code to FontPairPicker, restructure with consistent naming

Commit:
apply other block layouts

Commit:
init

Commit:
change request part 1

Arukuen committed:
feat: separate pro features

Arukuen committed:
feat: separate pro features

Arukuen committed:
fix: proper resetting in editingMode and in not

Arukuen committed:
fix: proper resetting in editingMode and in not

Arukuen committed:
feat: add confirmation when changing font pair while dirty

Arukuen committed:
feat: add confirmation when changing font pair while dirty

Commit:
styling tweaks

Commit:
styling tweaks

Commit:
Merge branch 'feat/3448-global-font-pairs' of https://github.com/gambitph/Stackable into feat/3448-global-font-pairs

# Conflicts:
#	src/components/font-pair-picker/index.js

Arukuen committed:
fix: make typography settings headings consistent

Arukuen committed:
fix: make typography settings headings consistent

Arukuen committed:
fix: error when no previously selected font pair

Arukuen committed:
fix: error when no previously selected font pair

Arukuen committed:
fix: pro notice update, translation, renaming default theme

Arukuen committed:
fix: pro notice update, translation, renaming default theme

Commit:
change request part 2

-hover state notice
-orange indicator
-reset button
-bug fixes

Arukuen committed:
fix: move editingMode to pro code

Arukuen committed:
fix: move editingMode to pro code

Arukuen committed:
fix: error when resetting

Arukuen committed:
fix: error when resetting

Commit:
change block components placeholder

Arukuen committed:
fix: reset display bug

Arukuen committed:
fix: reset display bug

Commit:
fix css

Arukuen committed:
Merge branch 'feat/3448-global-font-pairs' of https://github.com/gambitph/Stackable into feat/3448-global-font-pairs

Arukuen committed:
Revert "Merge branch 'feat/3448-global-font-pairs' of https://github.com/gambitph/Stackable into feat/3448-global-font-pairs"

This reverts commit 52ca1c16c1cd02b3e9e61f53b53781adfc995b63, reversing
changes made to 889ed65f78b2990028c9889bbc39b9d499d847d9.

Arukuen committed:
fix: pro notice button

Commit:
separate into spacing and borders, buttons and icons

Commit:
fix code

Commit:
rename functions

Commit:
fix four range control initial position

Commit:
border width in block editor override default

Commit:
include defaults.json

Commit:
add script for animation if block layout has hover

Commit:
fix bug (border width and border type)

Commit:
Merge branch 'feat/global-block-layout' into feat/global-color-schemes

# Conflicts:
#	src/plugins/global-settings/index.js

Commit:
add import

Commit:
Merge branch 'feat/global-block-layout' into feat/global-color-schemes

kaeizen committed:
Merge remote-tracking branch 'origin/feat/3448-global-font-pairs' into feat/global-color-schemes

Commit:
tweaks to the reset all button

kaeizen committed:
fix borderWidth override

kaeizen committed:
fix breakpoints

- use only 1 hook
- fix typography breakpoint issue

kaeizen committed:
fix hover animations, use stackable_frontend_css

kaeizen committed:
add desc and tooltips

kaeizen committed:
fix bug with shadow colors that uses rgba

Commit:
tweaked and rearranged settings. added correct docs. updated text

kaeizen committed:
use block-design-system.json

kaeizen committed:
remove defaults.json from buildInclude in gulp

kaeizen committed:
add documentation

kaeizen committed:
Merge branch 'feat/global-block-layout' into feat/global-color-schemes

kaeizen committed:
custom icons

kaeizen committed:
update block-design-system files

kaeizen committed:
create array/object for four-range values

kaeizen committed:
default color schemes

- add color scheme attributes
- save default color schemes to database

kaeizen committed:
Merge branch 'feat/global-block-layout' into feat/global-color-schemes

kaeizen committed:
add presets, proControl, filters for premium

kaeizen committed:
fix missing cssvar

kaeizen committed:
Merge branch 'feat/global-block-layout' into feat/global-color-schemes

kaeizen committed:
add reset and delete func, rearranged stuff

kaeizen committed:
fix block color scheme selector

kaeizen committed:
update pro notice

kaeizen committed:
init apply color scheme styles

kaeizen committed:
apply color schemes to editor and frontend

kaeizen committed:
remove bg color for :root

kaeizen committed:
fix typo

kaeizen committed:
fix color schemes if empty

kaeizen committed:
update comment

kaeizen committed:
update preset picker, add duplicate fx

kaeizen committed:
show color scheme colors in picker

kaeizen committed:
Merge branch 'develop' into feat/global-color-schemes

kaeizen committed:
fix add function, fix frontend if default is empty

kaeizen committed:
only add colors that have value in the picker

kaeizen committed:
move toggle to color palette, rearrange settings

kaeizen committed:
Merge branch 'develop' into feat/global-color-schemes

kaeizen committed:
update global colors test

kaeizen committed:
update global-settings test

kaeizen committed:
hover state

kaeizen committed:
update toggle in color palette

kaeizen committed:
update color scheme preview and color picker

kaeizen committed:
fix preview in edit

kaeizen committed:
fix color palette controls

kaeizen committed:
update editor loader

kaeizen committed:
apply parent-hover styles on frontend, add theme compatibility

kaeizen committed:
hover states

kaeizen committed:
fix array_map function

kaeizen committed:
fix other array_map functions

kaeizen committed:
fix theme compatibility

kaeizen committed:
add more presets, move theme compatibility code

kaeizen committed:
add documentation

Commit:
Merge branch 'develop' into feat/global-color-schemes

bfintal commented on src/hooks/use-block-color-schemes.js:
```diff
@@ -0,0 +1,136 @@
+
+		const allColorSchemes = applyFilters( 'stackable.global-settings.global-color-schemes.custom-color-schemes', colorSchemes, true )
+		const COLOR_SCHEME_OPTIONS = [ {
+			label: 'Scheme unavailable', // This will be displayed when a custom color scheme is deleted or when a license gets deactivated
```
Please make this text translatable

bfintal commented on src/hooks/use-block-color-schemes.js:
```diff
@@ -0,0 +1,136 @@
+		const toKebab = ( property, slug ) => {
+			const result = property.replace( /([a-z0-9])([A-Z])/g, '$1-$2' )
+			return '--stk-' + slug + '-' + result.toLowerCase()
+		}
```
Instead of doing this use lodash's `kebabCase` method

bfintal commented on src/block-components/container-div/index.js:
```diff
@@ -38,6 +39,7 @@ export const ContainerDiv = props => {
 		'stk-hover-parent': attributes.hasContainer && attributes.triggerHoverState, // This is needed to trigger parent-hover hover styles.
 		'stk--no-background': ! attributes.hasContainer,
 		'stk--no-padding': ! attributes.hasContainer,
+		[ `container-${ attributes.containerColorScheme }` ]: attributes.hasContainer && attributes.containerColorScheme,
```
Let's prefix this, please apply to the background also
```suggestion
		[ `stk--container-scheme--${ attributes.containerColorScheme }` ]: attributes.hasContainer && attributes.containerColorScheme,
```

bfintal commented on src/components/pro-control/index.js:
```diff
@@ -124,6 +124,15 @@ const LABELS = {
 		</ul>,
 	},
+	'color-schemes': {
+		title: __( 'Unlock Unlimited Color Schemes', i18n ),
```
```suggestion
		title: __( 'Premium Color Schemes', i18n ),
```

bfintal commented on src/plugins/global-settings/color-schemes/index.php:
```diff
@@ -0,0 +1,505 @@
+		 * @return Array
+		 */
+		public function convert_to_assoc_array( $schemes_array ) {
+			$schemes = array();
+
+			foreach( $schemes_array as $scheme ) {
+				$schemes[ $scheme['key'] ] = $scheme[ 'colorScheme' ];
+			}
+
+			return $schemes;
```
We can get rid of the foreach here:
```suggestion
			return array_column( $schemes_array, 'colorScheme', 'key' );
```

bfintal commented on src/plugins/global-settings/color-schemes/index.php:
```diff
@@ -0,0 +1,505 @@
+		 * @return String
+		 */
+		public function add_global_color_schemes_styles( $current_css ) {
+			$schemes_array = is_array( get_option( 'stackable_global_color_schemes' ) ) ? get_option( 'stackable_global_color_schemes' ) : [];
```
Is there a way to cache this so we won't have to regenerate this all the time?

bfintal commented on src/plugins/global-settings/color-schemes/index.php:
```diff
@@ -0,0 +1,505 @@
+			$schemes_array = is_array( get_option( 'stackable_global_color_schemes' ) ) ? get_option( 'stackable_global_color_schemes' ) : [];
+
+			// Get all color schemes, including custom color schemes if any
+			$all_color_schemes = apply_filters( 'stackable_global_color_schemes.get_color_schemes', $schemes_array );
```
For the filter, let's use `/` instead of `.` for the namespacing
```suggestion
			$all_color_schemes = apply_filters( 'stackable_global_color_schemes/get_color_schemes', $schemes_array );
```

bfintal committed:
Update src/components/pro-control/index.js [skip ci]

Commit:
fix for moving background image on hover

Commit:
Merge branch 'feat/global-color-schemes' of https://github.com/gambitph/Stackable into feat/global-color-schemes

Commit:
remove static from function

kaeizen committed:
minor fixes

translatable text,
use lodash's kebabCase,
add prefix,
use array_column,
rename namespace

kaeizen committed:
cache color scheme css
